pkg/image/unpacker/squashfs.go

```diff
@@ -62,7 +62,10 @@ type Squashfs struct {
 // NewSquashfs initializes and returns a Squahfs unpacker instance
 func NewSquashfs() *Squashfs {
 	s := &Squashfs{}
+	defaultPath := os.Getenv("PATH")
+	os.Setenv("PATH", os.Getenv("USER_PATH"))
```

Comment:
I think this is causing this unit tests to fail as `USER_PATH` isn't being set whereas it always is outside of the unit tests. I could workaround the issue by adding this to the test:

```go
os.Setenv("USER_PATH", os.Getenv("PATH"))
```

but I'll let someone take a look at this PR first to see if there is a better solution.

Reply:
Ideally we'd pass in something to the function if we need to modify PATH here, rather than rely on the `USER_PATH` env var being set here. I believe it is only set for the `actions` (run/exec/shell) and not builds.

```
09:06 AM $ rg "USER_PATH"
cmd/internal/cli/actions.go
50:	os.Setenv("USER_PATH", userPath)
```

The hope is we can stick more to using our custom env vars directly only in the CLI layer. See e.g. how the GPU code passes it down into functions that heed it...

```
# cmd/internal/cli/actions_linux.go

                // bind persistenced socket if found
                ipcs = gpu.NvidiaIpcsPath(userPath)
                libs, bins, err = gpu.NvidiaPaths(gpuConfFile, userPath)
```

Reply:
Thanks for taking a look, that makes a lot of sense. Changes made and the unit tests are now passing.